src/manifolds/StiefelSubmersionMetric.jl

```diff
@@ -0,0 +1,290 @@
+@doc raw"""
+    StiefelSubmersionMetric{T<:Real} <: RiemannianMetric
+
+The submersion (or normal) metric family on the [`Stiefel`](@ref) manifold.
+
+# Constructor
+
+    StiefelSubmersionMetric(α)
+
+Construct the submersion metric on the Stiefel manifold with the parameter ``α > -1``.
+
+The submersion metric family has two special cases:
+- ``α = -\frac{1}{2}``: [`EuclideanMetric`](@ref)
+- ``α = 0``: [`CanonicalMetric`](@ref)
+"""
+struct StiefelSubmersionMetric{T<:Real} <: RiemannianMetric
+    α::T
+end
+
+struct StiefelShootingInverseRetraction{
+    T<:Real,
+    R<:AbstractRetractionMethod,
+    VT<:AbstractVectorTransportMethod,
+} <: AbstractInverseRetractionMethod
+    max_iterations::Int
+    tolerance::T
+    num_transport_points::Int
+    retraction::R
+    vector_transport::VT
+end
+function StiefelShootingInverseRetraction(;
+    max_iterations=1_000,
+    tolerance=sqrt(eps()),
+    num_transport_points=4,
+    retraction=ExponentialRetraction(),
+    vector_transport=ScaledVectorTransport(ProjectionTransport()),
+)
+    return StiefelShootingInverseRetraction(
+        max_iterations,
+        tolerance,
+        num_transport_points,
+        retraction,
+        vector_transport,
+    )
+end
+
+struct StiefelKShootingInverseRetraction{T<:Real} <: AbstractInverseRetractionMethod
```

Comment:
Same as above + what is the `K` for?

Reply:
No this one is definitely Stiefel-specific. This is Algorithm 2 of the paper, and it's based on the observation that the solution and all intermediates of the shooting method for Stiefel occupy a `k \times k` matrix subspace, so when `n >> k`, this is much more efficient.

This is also why it's limited to a specific choice of vector transport and retraction. While for shooting we can be generic, here we cannot.

Reply:
Ah ok, just in reading this in use (`retract(M::Stiefel, p, X, m::StiefelKShootingInverseretraction)`stifle appears double now (and the `K` reads a little strange in the name.